aws-logs-loggroup/dev-resources.yaml

```diff
@@ -0,0 +1,39 @@
+AWSTemplateFormatVersion: "2010-09-09"
+Description: Development dependencies for AWS::Logs::LogGroup
+Resources:
+  Cmk:
+    Type: AWS::KMS::Key
+    Properties:
+      Description: KMS CMK for use with the AWS::Logs::LogGroup resource provider
+      KeyUsage: ENCRYPT_DECRYPT
+      KeyPolicy:
+        Version: 2012-10-17
+        Id: cmk-policy
+        Statement:
+        - Sid: Enable IAM User Permissions
+          Effect: Allow
+          Principal:
+            AWS: !Sub "arn:${AWS::Partition}:iam::${AWS::AccountId}:root"
+          Action: kms:*
+          Resource: "*"
+        - Sid: CloudWatch Logs Permissions
+          Effect: Allow
+          Principal:
+            Service: !Sub "logs.${AWS::Region}.${AWS::URLSuffix}"
+          Action:
+            - "kms:Encrypt*"
+            - "kms:Decrypt*"
+            - "kms:ReEncrypt*"
+            - "kms:GenerateDataKey*"
+            - "kms:Describe*"
+          Resource: "*"
+  # Alias is only added to show the StackName in the KMS Console
+  CmkAlias:
+    Type: "AWS::KMS::Alias"
+    Properties:
+      AliasName: !Sub "arn:${AWS::Partition}:kms:${AWS::Region}:${AWS::AccountId}:alias/${AWS::StackName}"
```

Comment:
@ikben this is giving me 

```
"ResourceStatus": "CREATE_FAILED",
"ResourceStatusReason": "Model validation failed (#/AliasName: failed validation constraint for keyword [pattern])"
```

https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-kms-alias.html#aws-resource-kms-alias-properties

Reply:
@miparnisari Thanks for the fix. Yours does follow what's in the documentation.

At the time of writing, it did deploy though (I checked the deleted stack in my account, it has the long !Sub). So make sure that it's not the StackName that is causing the failure.